src/main/java/org/jlab/mya/stream/MySamplerStream.java

```diff
@@ -0,0 +1,152 @@
+ * interval in terms of time, and the number of samples, along with an EventStream and prior Event that contains the data
+ * to be sampled from.
+ * @param <T> The Event type (FloatEvent, IntEvent, etc.)
+ */
+public class MySamplerStream<T extends Event> extends BoundaryAwareStream<T> {
+
+    private final long intervalMillis;
+    private final long sampleCount;
+    private long samplesTaken = 0L;
+    private long sampleTimeMya;
+    private Instant sampleTimeInstant;
+    private T currentEvent = null;
+    private T previousEvent = null;
+    private boolean firstRead = true;
+    private boolean endOfStream = false;
+
+    /**
+     * Create an instance of MySamplerStream.  This extends a special case of the BoundaryAwareStream that always has
+     * priorPoint supplied.  The BoundaryAwareStream has 'begin' and 'end' values that match the first and last sample
+     * needed in the mySamplerStream.
+     *
+     * @param wrapped        A stream of Events to be sampled from
+     * @param begin          The time of the first sample
+     * @param intervalMillis The time interval between samples in milliseconds
+     * @param sampleCount    The number of samples to take, including the first one at begin
+     * @param priorPoint     The Event prior to begin.  It's required to be non-null.
+     * @param updatesOnly    Should the final event from the BoundaryAwareStream only be an update event.
+     * @param type           The type of Event produced by the wrapped stream
+     */
+    public MySamplerStream(EventStream<T> wrapped, Instant begin, long intervalMillis, long sampleCount, T priorPoint,
+                           boolean updatesOnly, Class<T> type) {
+        super(wrapped, begin, begin.plusMillis(intervalMillis * (sampleCount - 1)), priorPoint, updatesOnly,
+                type);
+        if (priorPoint == null) {
+            throw new IllegalArgumentException("mySamplerStream requires a non-null priorPoint.");
+        }
+        this.intervalMillis = intervalMillis;
+        this.sampleCount = sampleCount;
+        this.sampleTimeMya = TimeUtil.toMyaTimestamp(begin);
+        this.sampleTimeInstant = begin;
+    }
+
+    /**
+     * This does in app sampling of the full data stream.  The basic idea is to return the value of the signal only at
+     * the times that were requested based on sampling parameters.  Events are created so that timestamps are at sample
+     * times.  null is returned when requesting the future.
+     * <p>
+     * We maintained two event update points.  When they straddle the sample time, we read out an event with the sample
+     * time timestamp, but the value and code of the previous update event.  Each read from this stream will iterate
+     * over the wrapped stream until we find two points that straddle the sample time or the end of the stream.
+     *
+     * @return The next sample event or null if future.
+     * @throws IOException If unable to read the next event
+     */
```

Comment:
Add a `@SuppressWarnings("unchecked")` annotation to the read method to avoid compiler warnings.

Reply:
Done.